### PR TITLE
Reduced default obstacle queue cap to mitigate CORE-1665

### PR DESCRIPTION
**Pull request by jasonimercer**
Real solution will involve a data structure change. This will happen in the next sprint under CORE-1669.

### CODE REVIEW

**Comment by jasonimercer** [comment]
@skaynama @abencz 

New ticket, CORE-1669 to address this in a smarter way for the next sprint.

**Comment by skaynama** [comment]
Can you explain why this would solve CORE-1665? 
Also, + @servos .

**Comment by jasonimercer** [comment]
I don't think it will solve it but it will make it less of a problem. I think the 200% CPU is from the two obstacle layer threads iterating over their internal lists looking for duplicates at every iteration. The current max of 50k obstacles means we have approx (50k)^2 compares when it has filled up. This is horrible. Reducing max to 5k (and so "only" 5k^2 compares) will allow it to complete faster. New code removes this problem but will require testing we don't have time for today.

**Comment by jasonimercer** [comment]
More changes have crept into this branch:

We will now only remember observations (to be explicitly forgotten if outside remember radius) if we have confidence in our pose.

File: costmap_2d/plugins/obstacle_layer.cpp [hunk 1]
```diff
@@ -67,6 +67,13 @@ void ObstacleLayer::onInitialize()
   obstacle_compare_tolerance_ = 0.0; // meters
   use_forgetful_version_ = true;     // flag
 
+  // Initial pose confidence and threshold before we remember new data
+  // Threshold set in costmap_2d/cfg/ObstaclePlugin.cfg
+  pose_confidence_ = 0;
+  pose_confidence_threshold_ = 1;
+
+  pose_confidence_sub_ = g_nh.subscribe("/slam/scan_match_score", 1, &ObstacleLayer::poseConfidenceCallback, this);
```
Comment (skaynama) [inline costmap_2d/plugins/obstacle_layer.cpp]: I think the leading "/" should be removed.

**Comment by skaynama** [comment]
LGTM

File: costmap_2d/plugins/obstacle_layer.cpp (hunk 1 above)
Comment (jasonimercer) [inline costmap_2d/plugins/obstacle_layer.cpp]: good catch. Thanks.

**Pull request closed without merging**